ddtrace/propagation/http.py

```diff
@@ -182,10 +181,17 @@ def _extract(headers):
             # Do not fail if the tags are malformed
             try:
                 # We get a Dict[str, str], but need it to be Dict[Union[str, bytes], str] (e.g. _MetaDictType)
-                meta = cast(Dict[Union[str, bytes], str], decode_tagset_string(tags_value))
+                meta = cast(
+                    Dict[Union[str, bytes], str],
+                    {k: v for (k, v) in decode_tagset_string(tags_value) if k in _DatadogMultiHeader._PROPAGATED_TAGS},
+                )
             except TagsetDecodeError:
                 log.debug("failed to decode x-datadog-tags: %r", tags_value, exc_info=True)
 
+        # Parse tags even if propagation is disabled
+        if not config._propagation_datadog_tags_enabled:
+            meta = None
```

Comment:
I think this can go away? it doesn't match the comment.

Reply:
We don't skip the previous decoding of tagset and reset the meta if this feature is disabled. I can clarify that in the comment.